### ISSUE TITLE
10 band Equalizer with Airplay/Bluetooth streaming from smartphone

### ISSUE THREAD

**Issue by rsm123** [sle118/squeezelite-esp32#109]
Thanks for this wonderful software! I'm trying to understand how to use the 10 band Equalizer while streaming Airplay or Bluetooth from an iphone or Android phone. Not sure if I'm missing anything but I could not find any info related to setting Equalizer in the Readme or the Web interface. Is this feature usable at this time?

**Comment by sle118** [comment]
Currently, the equalizer only works when connected to the LMS. To get it to work when in "offline mode" persistence needs to be implemented.

**Comment by philippe44** [comment]
@sle118: please review the use of nvs_xxx. I can't use platform_config as I need blob but I don't remember the differences with nvs_xxx. Is it the delayed write?

**Comment by sle118** [comment]
> @sle118: please review the use of nvs_xxx. I can't use platform_config as I need blob but I don't remember the differences with nvs_xxx. Is it the delayed write?

Funny enough I had edited your comment instead of replying (didn't know this was possible). So here was my response

I have reviewed the code, and yes the difference is the timer decoupling that allowed saving to Flash from a psram stack without smashing the cache. Let's keep the code as is for now, unless you think we're going to cause cache issues. I will update the method later and implement a mechanism that will allow using the config subsystem by covering to and from a hex string internally, or with a better approach

**Comment by philippe44** [comment]
There is no risk at all for that to be used in PSRAM. You don't even need to worry about changing that then

**Comment by sle118** [comment]
@rsm123 once the equalizer has been configured in LMS, it should now be saved and work with Bluetooth thanks to @philippe44 with commit https://github.com/sle118/squeezelite-esp32/commit/48e8525ba97657c155167095a7f0361146c64ebb

I'll close this for now

**Comment by larryqiann** [comment]
Hi,

Unsure where to put this. I'm using the latest build of Squeezelite-ESP32 and the newest Lyrion Music Server with the SqueezeESP32 plugin enabled in settings, but I can't find the equalizer option. What's the procedure for finding and using this functionality? 

Thanks!

**Comment by philippe44** [comment]
You need to install the Lyrion associated plug-in

**Comment by larryqiann** [comment]
![Image](https://github.com/user-attachments/assets/d5bf6c28-e887-4f8c-ac7a-6fe9a727a3ff)

I've installed this and restarted LMS already. I'm assuming I should be looking under Player > Audio settings but I don't see the options.

Thanks for your quick reply!

**Comment by philippe44** [comment]
There are some rare cases that I've not been able to identify where the esp32 device is not proprely detected. Try to erase the player using Michael's plugin (can't remember the name, but it's made to erase inactive devices) and connect it again.

**Comment by larryqiann** [comment]
I've just used the ESP web installer tool to reflash and erase the device, as well as factory resetting LMS. 
I do have a menu called "ESP32 settings" in the menu, but there is no equalizer config options under it. Could you confirm what the control should look like on a working setup? (Have tried both default and material skins)

I'm using the Mac version of LMS,  Version: 9.0.2 - 1741501853 

![Image](https://github.com/user-attachments/assets/3e6d590e-51a5-488f-99e8-a10e2c1ef065)

**Comment by philippe44** [comment]
oh, are you using the 32 bits version ? There is no equalizer then

**Comment by larryqiann** [comment]
I'm using the I2S-4MFLASH-16 variant, so that should be the 16 bit version if I'm remembering correctly. I'm trying to reset/connect it to a Windows LMS instead to see if the problem persists. (is there an easy way to unpair the player without having to erase it and reflash? I'm having a hard time getting it to connect even after erasing)

**Comment by philippe44** [comment]
Reflashing does not do anything for that issue. The only thing that might help is to remove the player form LMS

**Comment by larryqiann** [comment]
What is the easiest way to remove a player from LMS? I couldn't get the player to connect to Windows at all, but the same problem is experienced on another Mac.


EDIT: magically it started to work on the second Mac. I'm not sure what I did, but I did open the devtools console in Chrome to see what might have been happening, and without doing anything the settings show up now. 

Under Player Settings, they appear under Extra Player Settings > Plugins > ESP32 settings
